contrib/pkg/installmanager/installmanager.go

```diff
@@ -239,6 +237,19 @@ func (m *InstallManager) uploadClusterMetadata() error {
 			"metadata.json": string(metadataBytes),
 		},
 	}
+
+	cd := &hivev1.ClusterDeployment{}
+	err = m.DynamicClient.Get(context.Background(), types.NamespacedName{Namespace: m.Namespace, Name: m.ClusterName}, cd)
+	if err != nil {
+		m.log.WithError(err).Error("error getting cluster deployment")
+		return err
+	}
+	m.log.Debug("got cluster deployment")
+	if err := controllerutil.SetControllerReference(cd, metadataCfgMap, scheme.Scheme); err != nil {
```

Comment:
shouldn't you also set the controller reference on the kubeconfig secret?

Reply:
Oops!